### PR TITLE
feat: Cancellation siri

### PR DESCRIPTION
**Pull request by alainajaffer**

### CODE REVIEW

File: packages/siri-sx-importer/utils/importerSiriTypes.zod.ts [hunk 1]
```diff
@@ -126,14 +126,25 @@ export const stopPointsSchema = z.object({
...
+            VehicleJourneyRef: z.string(),
+            Route: z.string(),
+            OriginAimedDepartureTime: z.string(),
+        }),
+    ),
+});
+
 export const consequenceItem = z.object({
-    Condition: z.literal("unknown"),
+    Condition: z.literal("unknown").or(z.literal("cancelled")),
```
Comment (DeannaSharma) [inline packages/siri-sx-importer/utils/importerSiriTypes.zod.ts:140]:
```suggestion
    Condition: z.enum(["unknown", "cancelled"]),
```
```suggestion
    Condition: z.literal("unknown").or(z.literal("cancelled")),
```

File: packages/siri-sx-generator/util/siri.ts [hunk 2]
```diff
@@ -149,6 +154,7 @@ export const getPtSituationElementFromSiteDisruption = (
                                     }
                                   : {}),
                               ...(consequence.consequenceType === "networkWide" ||
+                              (consequence.consequenceType === "journeys" && CANCELLATION_FEATURE_FLAG) ||
```
Comment (DeannaSharma) [inline packages/siri-sx-generator/util/siri.ts:157]: According to the ticket, if we have a journey type consequence then we should specify the Line that is affected - we should be able to access this in the same way as we did for operators by mapping over the consequence.service.

I would check with Steven if we're implementing the LineRef changes we made in earlier tickets here i.e LineRef = lineId and PublishedLineName = lineName

File: packages/siri-sx-generator/util/siri.ts
Comment (DeannaSharma) [inline packages/siri-sx-generator/util/siri.ts:1]: When we generate the csv file as part of this function, we do a count of the servicesAffected, stopsAffected etc. We should ask Steve if we need to update this to count the journeysAffected or whether we will do this in a separate ticket

Screenshot 2024-07-04 at 07 40 48

File: shared-ts/siriTypes.zod.ts [hunk 3]
```diff
@@ -124,16 +124,27 @@ export const stopPointsSchema = z.object({
...
+            OriginAimedDepartureTime: z.string(),
+        }),
+    ),
+});
+
 export const consequenceSchema = z.object({
     Consequence: z.array(
         z.object({
-            Condition: z.literal("unknown"),
+            Condition: z.literal("unknown").or(z.literal("cancelled")),
```
[the remark this answers is not in the document]
Comment (alainajaffer) [reply shared-ts/siriTypes.zod.ts]: done

File: packages/siri-sx-importer/utils/importerSiriTypes.zod.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (alainajaffer) [reply packages/siri-sx-importer/utils/importerSiriTypes.zod.ts]: done

File: packages/siri-sx-generator/util/siri.ts
Comment (alainajaffer) [reply packages/siri-sx-generator/util/siri.ts:1]: a separate ticket

File: packages/siri-sx-generator/util/siri.ts (hunk 2 above)
Comment (alainajaffer) [reply packages/siri-sx-generator/util/siri.ts]: done

File: packages/siri-sx-generator/util/siri.ts
Comment (DeannaSharma) [reply packages/siri-sx-generator/util/siri.ts:1]: ticker DISRUPTION-749 made

File: shared-ts/siriTypes.zod.ts
Comment (DeannaSharma) [inline shared-ts/siriTypes.zod.ts:1]: Currently with how we have the network schema set up the generator is failing to run due to zod errors.

On line 84 update the AffectedOperator element to be optional:
`AffectedOperator: affectedOperatorSchema.optional(),`

Comment (alainajaffer) [reply shared-ts/siriTypes.zod.ts:1]: done

**Review by DeannaSharma — APPROVED** [review]

**Pull request merged**